File: past/B_Can_Sorting_Fail.cpp
#include <bits/stdc++.h>

typedef long long ll;
typedef long double ld;
using namespace std;

const char ENDL = '\n';
const ll MOD = 1e9 + 7;

/*
3
3
2 2 1
4
3 1 2 1
5
1 2 2 4 4
*/

void solve()
{
    ll n;
    cin >> n;
    vector<ll> nums(n, 0);
    for (int i = 0; i < n; i++) {
        cin >> nums[i];
    }
    vector<ll> mines(n), maxes(n);
    maxes[0] = nums[0];
    mines[n - 1] = nums[n - 1];
    for (int i = 1; i < n; i++) {
        maxes[i] = max(maxes[i - 1], nums[i]);
    }
    for (int j = n - 2; j >= 0; j--) {
        mines[j] = min(mines[j + 1], nums[j]);
    }
    bool found = false;
    for (int k = 1; k < n && !found; k++) {
        if (maxes[k - 1] <= mines[k]) {
            found = true;
        }
    }

    if (found) {
        cout << "NO" << ENDL;
    } else {
        cout << "YES" << ENDL;
    }
}

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    int t;
    std::cin >> t;
    while (t--)
        solve();
    return 0;
}

File: past/B_Can_Sorting_Fail.py
t = int(input())

for _ in range(t):
    n = int(input())
    nums = list(map(int, input().split()))
    mins = [nums[n - 1] for i in range(n)]
    maxs = [nums[0] for i in range(n)]
    maxs[0] = nums[0]
    mins[n - 1] = nums[n - 1]
    for i in range(1, n):
        maxs[i] = max(maxs[i - 1], nums[i])
    for j in range(n - 2, -1, -1):
        mins[j] = min(mins[j + 1], nums[j])
    found = False
    for i in range(1, n):
        if maxs[i - 1] <= mins[i]:
            found = True
            break
    print("NO" if found else "YES")